testing/scenario_app/ios/Scenarios/ScenariosUITests/PlatformViewUITests.m

```diff
@@ -0,0 +1,118 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#import <Flutter/flutter.h>
+#import <XCTest/XCTest.h>
+#include <sys/sysctl.h>
+
+#import "../Scenarios/TextPlatformView.h"
+
+@interface PlatformViewUITests : XCTestCase
+@property(nonatomic, strong) XCUIApplication* application;
+@end
+
+@implementation PlatformViewUITests
+
+- (void)setUp {
+  [super setUp];
+  self.continueAfterFailure = NO;
+
+  self.application = [[XCUIApplication alloc] init];
+  self.application.launchArguments = @[ @"--platform-view" ];
+  [self.application launch];
+}
+
+- (void)testPlatformView {
+  NSBundle* bundle = [NSBundle bundleForClass:[self class]];
+  NSString* goldenName =
+      [NSString stringWithFormat:@"golden_platform_view_%@", [self platformName]];
+  NSString* path = [bundle pathForResource:goldenName ofType:@"png"];
+  UIImage* golden = [[UIImage alloc] initWithContentsOfFile:path];
+
+  XCUIScreenshot* screenshot = [[XCUIScreen mainScreen] screenshot];
+  if (golden) {
+    XCTAttachment* goldenAttachment = [XCTAttachment attachmentWithImage:golden];
+    goldenAttachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+    [self addAttachment:goldenAttachment];
+  }
+
+  XCTAttachment* attachment = [XCTAttachment attachmentWithScreenshot:screenshot];
+  attachment.lifetime = XCTAttachmentLifetimeKeepAlways;
+  [self addAttachment:attachment];
+  XCTAssertTrue([self compareImage:golden toOther:screenshot.image]);
+}
+
+- (NSString*)platformName {
+  NSString* simulatorName =
+      [[NSProcessInfo processInfo].environment objectForKey:@"SIMULATOR_DEVICE_NAME"];
+  if (simulatorName) {
+    return [NSString stringWithFormat:@"%@_simulator", simulatorName];
+  }
+
+  size_t size;
+  sysctlbyname("hw.model", NULL, &size, NULL, 0);
```

Comment:
Bleh I thought this was deprecated.
What did -[UIDevice platformString] return (I think you tried that first).

Reply:
That was a category someone made that delegates to this, more or less.  This is probably not safe for stores but I only need it for tests.